src/plugins/vis_type_vega/public/expressions/helpers.ts

```diff
@@ -0,0 +1,279 @@
+  const updatedRows: OpenSearchDashboardsDatatableRow[] = datatable.rows.map((row) =>
+    Object.entries(row).reduce((updatedRow, [key, value]) => {
+      const cleanKey = key.replaceAll('.', '-');
+      return Object.assign(updatedRow, { [cleanKey]: value });
+    }, {})
+  );
+
+  datatable.rows = updatedRows;
+  return datatable;
+};
+
+export const setupConfig = (visParams: VisParams, config: VisAugmenterEmbeddableConfig) => {
+  const legendPosition = visParams.legendPosition;
+  return {
+    view: {
+      stroke: null,
+    },
+    concat: {
+      spacing: 0,
+    },
+    legend: {
+      orient: legendPosition,
+      offset: calculateLegendOffset(legendPosition),
+    },
+    // This is parsed in the VegaParser and hides unnecessary warnings.
+    // For example, 'infinite extent' warnings that cover the chart
+    // when there is empty data for a time series
+    kibana: {
+      hideWarnings: true,
+      visAugmenterConfig: config,
+    },
+  };
+};
+
+export const buildLayerMark = (seriesParams: {
+  type: string;
+  interpolate: string;
+  lineWidth: number;
+  showCircles: boolean;
+}) => {
+  return {
+    // Possible types are: line, area, histogram. The eligibility checker will
+    // prevent area and histogram (though area works in vega-lite)
+    type: seriesParams.type,
+    // Possible types: linear, cardinal, step-after. All of these types work in vega-lite
+    interpolate: seriesParams.interpolate,
+    // The possible values is any number, which matches what vega-lite supports
+    strokeWidth: seriesParams.lineWidth,
+    // this corresponds to showing the dots in the visbuilder for each data point
+    point: seriesParams.showCircles,
+  };
+};
+
+export const buildXAxis = (xAxisTitle: string, xAxisId: string, visParams: VisParams) => {
+  return {
+    axis: {
+      title: xAxisTitle,
+      grid: visParams.grid.categoryLines,
+    },
+    field: xAxisId,
+    // Right now, the line charts can only set the x-axis value to be a date attribute, so
+    // this should always be of type temporal
+    type: 'temporal',
+  };
+};
+
+export const buildYAxis = (
+  column: OpenSearchDashboardsDatatableColumn,
+  valueAxis: ValueAxis,
+  visParams: VisParams
+) => {
+  const subAxis = {
+    title: cleanString(valueAxis.title.text) || column.name,
+    grid: visParams.grid.valueAxis !== undefined,
+    orient: valueAxis.position,
+    labels: valueAxis.labels.show,
+    labelAngle: valueAxis.labels.rotate,
+  };
+  // Percentile ranks aggregation metric needs percentile formatting.
+  if (column.meta?.type === 'percentile_ranks') Object.assign(subAxis, { format: '.0%' });
+  return {
+    axis: subAxis,
+    field: column.id,
+    type: 'quantitative',
+  };
+};
+
+const isXAxisColumn = (column: OpenSearchDashboardsDatatableColumn): boolean => {
+  return column.meta?.aggConfigParams?.interval !== undefined;
+};
+
+export const createSpecFromDatatable = (
```

Comment:
I think we may have gone back and forth on the naming a bit in the feature branch, but there is a lot of assumptions in this function that are specific to temporal x-y visualizations, so we should explicitly name it so that's clear.

Reply:
Good point - I'll rename + add comment above the fn.